source/common/network/address_impl.cc

```diff
@@ -397,7 +405,14 @@ Api::SysCallIntResult PipeInstance::bind(int fd) const {
     unlink(address_.sun_path);
   }
   auto& os_syscalls = Api::OsSysCallsSingleton::get();
-  return os_syscalls.bind(fd, sockAddr(), sockAddrLen());
+  auto bind_result = os_syscalls.bind(fd, sockAddr(), sockAddrLen());
+  if (mode != 0 && !abstract_namespace_ && bind_result.rc_ == 0) {
+    auto set_permissions = os_syscalls.chmod(address_.sun_path, mode);
+    if (set_permissions.rc_ != 0) {
+      throw EnvoyException(fmt::format("Failed to create socket with mode {}", mode));
```

Comment:
Do you have a test for this case? Same with all the other exception cases?

Reply:
added [test](https://github.com/envoyproxy/envoy/pull/8423/files#diff-a97b9874a5d6e4ba7ea09da95a425b58R345) to cover this. The abstract socket case is already covered.